tools/hsmtool.c

```diff
@@ -492,7 +491,8 @@ int main(int argc, char *argv[])
 
 		return dumponchaindescriptors(argv[2],
 					      argc > 3 && !streq(argv[3], "") ? argv[3] : NULL,
-					      argc > 4 && streq(argv[4], "private"));
+					      argc > 4 && streq(argv[4], "private"),
+					      argc > 5 && streq(argv[5], "testnet"));
```

Comment:
same as previous comment, this seems like it should be checking for 'true/false' not "testnet"